client/state/data-layer/wpcom/sites/activity/from-api.js

```diff
@@ -20,13 +20,15 @@ export const DEFAULT_GRIDICON = 'info-outline';
 /**
  * Transforms API response into array of activities
  *
- * @param  {object} apiResponse                      API response body
- * @param  {array}  apiResponse.current.orderedItems Array of item objects
- * @return {array}                                   Array of proccessed item objects
+ * @param  {object} apiResponse API response body
+ * @return {object}             Object with an entry for proccessed item objects and another for oldest item timestamp
  */
 export function transformer( apiResponse ) {
-	const orderedItems = get( apiResponse, [ 'current', 'orderedItems' ], [] );
-	return map( orderedItems, processItem );
+	return {
+		items: map( get( apiResponse, [ 'current', 'orderedItems' ], [] ), processItem ),
+		oldestItemTs: get( apiResponse, [ 'oldestItemTs' ], 0 ),
+		totalItems: get( apiResponse, [ 'totalItems' ], 0 ),
```

Comment:
two things here: we can safely grab `apiResponse.totalItems` without `get()` because `apiResponse` is guaranteed to be an object since it already went through validation. also, I'm not sure that `0` is a safe default. what should we do in such a case? should we maybe pass `apiResponse.current.orderedItems.length`? should we toss an error into Tracks?

Reply:
Nice catch. I'll pass `apiResponse.current.orderedItems.length`.